File: ChatGPT/c/output/output_03a.c
// ChatGPT-generated solution will go here.
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>

int main() {
    FILE *f = fopen("input.txt", "r");
    char ch;
    int total = 0;
    
    while ((ch = fgetc(f)) != EOF) {
        if (ch == 'm') {
            if (fgetc(f) == 'u' && fgetc(f) == 'l' && fgetc(f) == '(') {
                int x = 0, y = 0;
                // Read first number
                while (isdigit(ch = fgetc(f))) {
                    x = x * 10 + (ch - '0');
                }
                // Skip comma
                fgetc(f);
                // Read second number
                while (isdigit(ch = fgetc(f))) {
                    y = y * 10 + (ch - '0');
                }
                // Skip closing parenthesis
                fgetc(f);
                total += x * y;
            }
        }
    }

    fclose(f);
    printf("%d\n", total);
    return 0;
}

File: ChatGPT/clojure/output/output_03a.clj
;; ChatGPT-generated solution will go here.
(ns output_03a
  (:require [clojure.string :as str]))

(defn part [lines]
  (let [input (apply str lines)
        pattern #"mul\((\d{1,3}),(\d{1,3})\)"]
    (->> (re-seq pattern input)
         (map (fn [[_ a b]] (* (parse-long a) (parse-long b))))
         (reduce +))))

(defn -main [& args]
  (let [filename (first args)
        lines (str/split-lines (slurp filename))]
    (println (part lines))))
